Sources/ProjectDescription/SettingsTransformers.swift

```diff
@@ -0,0 +1,147 @@
+import Foundation
+
+extension Dictionary {
+    mutating func merge(with dict: [Key: Value]) {
+        dict.forEach { self[$0.key] = $0.value }
+    }
+
+    func merging(with dict: [Key: Value]) -> Dictionary {
+        var copy = self
+        copy.merge(with: dict)
+        return copy
+    }
+}
+
+private extension Bool {
+    var asSettingValue: SettingValue {
+        self ? "YES" : "NO"
+    }
+}
+
+private extension String {
+    var asSettingValue: SettingValue {
+        .init(stringLiteral: self)
+    }
+}
+
+public enum SwiftCompilationMode: String {
+    case singlefile
+    case wholemodule
+}
+
+public enum SwiftOptimizationLevel: String {
+    case o = "-O"
+    case oNone = "-Onone"
+    case oSize = "-Osize"
+}
+
+public extension SettingsDictionary {
```

Comment:
It may be worth a few small tests that verify the final raw settings 

e.g.

```swift
func test_settings_chainingMultipleValues() {
   /// Given / When
   let settings = SettingsDictionary()
     .codeSignIdentityAppleDevelopment()
     .currentProjectVersion("1")
     // ...

   /// Then
   XCTAssertEqual(settings, [
        "CODE_SIGN_STYLE": "Apple Development",
        "CURRENT_PROJECT_VERSION": "1",
        // ...
    ])
}
```